R/checks.R

```diff
@@ -145,3 +145,77 @@ check_sparse_pmf_tail <- function(pmf, span = 5, tol = 1e-6) {
     )
   }
 }
+
+#' Check if data has either explicit NA values or implicit missing dates.
+#'
+#' @param data The data to be checked
+#'
+#' @return `TRUE` if data is complete, else if data has implicit or explicit
+#' missingness, `FALSE`.
+#' @keywords internal
+test_data_complete <- function(data) {
+  data <- setDT(data) # Convert data to data.table
+
+  # Check for explicit missingness in required columns
+  # (date, confirm, primary, secondary)
+  columns_to_check <- c(
+    "date",
+    intersect(c("confirm", "primary", "secondary"), names(data))
```

Comment:
Hardcoding these column names will require coming back to this when addressing #505 - perhaps we should just pass this as an argument (where the function asking for the tests specifies what to check)?

Reply:
This has been fixed in 8fc148e8111a6b402f9ea3508cabd160097e40a2.